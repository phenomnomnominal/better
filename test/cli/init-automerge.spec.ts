import { describe, it, expect } from 'vitest';

import { createFixture } from '../fixture.js';

const ARGV = ['node', './bin/betterer'];

describe('betterer cli', () => {
  it('should enable automerge configuration', async () => {
    const { cliÎ” } = await import('@betterer/cli');

    const { cleanup, logs, paths, resolve, readFile } = await createFixture(
      'init-automerge',
      {
        './package.json': `
      {
        "name": "init-automerge",
        "version": "0.0.1"
      }
      `,
        './.git/file.txt': 'HOLD'
      },
      {
        logFilters: [/ðŸŒŸ Initialising Betterer/]
      }
    );

    const fixturePath = paths.cwd;

    process.env.BETTERER_WORKER = 'false';

    await cliÎ”(fixturePath, [...ARGV, 'init', '--automerge']);

    const attributesFile = await readFile(resolve('.gitattributes'));

    expect(attributesFile).toMatchSnapshot();

    const gitconfigFile = await readFile(resolve('./.git/config'));

    expect(gitconfigFile).toMatchSnapshot();

    expect(logs).toMatchSnapshot();

    await cleanup();
  });
});
