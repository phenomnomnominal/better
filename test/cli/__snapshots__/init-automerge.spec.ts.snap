// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`betterer cli > should enable automerge configuration 1`] = `
"# Betterer merge
.betterer.results merge=betterer
"
`;

exports[`betterer cli > should enable automerge configuration 2`] = `
"[merge "betterer"]
	driver = ../../../../packages/cli/bin/betterer.js merge %A %B
"
`;

exports[`betterer cli > should enable automerge configuration 3`] = `
[
  "ðŸŽ‰ Initialising Betterer (0ms): 3 tasks done!
âœ… Create test file: done!
ãƒ» created "fixtures/init-automerge/.betterer.ts"!
âœ… Update package.json: done!
âœ… Enable automerge: done!
ãƒ» added Betterer merge config to "fixtures/init-automerge/.git/config"!
ãƒ» added Betterer merge attribute to "fixtures/init-automerge/.gitattributes"!",
  "ðŸŽ‰ Initialising Betterer (0ms): 3 tasks done!
âœ… Create test file: done!
ãƒ» created "fixtures/init-automerge/.betterer.ts"!
âœ… Update package.json: done!
ãƒ» added "betterer" script to package.json file
ãƒ» added "@betterer/cli" dependency to package.json file
ãƒ» added "typescript" dependency to package.json file
âœ… Enable automerge: done!
ãƒ» added Betterer merge config to "fixtures/init-automerge/.git/config"!
ãƒ» added Betterer merge attribute to "fixtures/init-automerge/.gitattributes"!",
]
`;
